pyquil/api/_qpu.py

```diff
@@ -86,6 +88,28 @@ def get_version_info(self) -> dict:
         """
         return self.client.call('get_version_info')
 
+
+    @_record_call
+    def load(self, executable):
+        """
+        Initialize a QAM into a fresh state. Load the executable and parse the expressions
+        in the recalculation table (if any) into pyQuil Expression objects.
+
+        :param executable: Load a compiled executable onto the QAM.
+        """
+        super().load(executable)
+
+        try:
+            recalculation_table = self._executable.recalculation_table
+            for memory_reference, recalc_rule in recalculation_table.items():
+                # Parse the arithmetic expression
+                expression = parse(f"RZ({recalc_rule}) 0")[0].params[0]
```

Comment:
can we make a note here that explains that this is a bit of a hack? e.g. the presence of the `RZ` is meaningless

Reply:
yes, good call out. very much looking forward to #687 being merged so we can replace these lines with something more sensible

Reply:
how would you do it with #687? can you not parse expressions directly with antlr?